src/server/pfs/s3/s3.go

```diff
@@ -109,20 +110,46 @@ func Server(port uint16, driver Driver, clientFactory ClientFactory) (*http.Serv
 	s3Server.Bucket = c
 	s3Server.Object = c
 	s3Server.Multipart = c
-	router := s3Server.Router()
+	return s3Server.Router()
+}
+
+// Server runs an HTTP server with an S3-like API for PFS. This allows you to
+// use s3 clients to access PFS contents.
 
+// Modifications to `routerMap` allow dynamic changes for the mapping between s3 endpoints and API configurations
+func Server(port uint16, defaultRouter *mux.Router, routerMap map[string]*mux.Router, routersLock *sync.Mutex) *http.Server {
```

Comment:
Is there a particular reason to pass the map and mutex in, rather than having the Server own them itself

Reply:
the reason is that ServeSidecarS3G is modifying that map concurrently to add/remove endpoint router objects. We could definitely push another layer of abstraction and have another Server wrapper object that manages this. 
What do you think?